src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexOpcode.cs

```diff
@@ -117,54 +117,169 @@ internal enum RegexOpcode
         UpdateBumpalong = 46,
 
         // Primitive control structures
-        // TODO: Figure out what these comments mean / what these control structures actually do :)
 
-        /// <summary>back     jump            straight first.</summary>
+        /// <summary>Lazy branch in an alternation or conditional construct.</summary>
+        /// <remarks>
+        /// On first execution, the opcode records the current input position (via the tracking stack) and continues straight
+        /// without taking the jump. When the matching that follows fails, backtracking will occur and the saved position is restored,
+        /// at which point the interpreter will jump to the alternative branch (using the patched jump offset in operand 0).
+        /// This opcode is used to implement alternation in a non-greedy (lazy) manner.
+        /// </remarks>
         Lazybranch = 23,
```

Comment:
nit: I realize this is an internal enum, but should probably use camel casing here are most other cases.